sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Scenarios/TestScenario.cs

```diff
@@ -1,17 +1,97 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+
 namespace Azure.Messaging.EventHubs.Stress;
 
 /// <summary>
-///   The set of test scenarios that can be run.
-/// </summary>
+///   This is the base class to implement when adding a new test scenario to the stress test suite.
+/// <summary/>
 ///
-public enum TestScenario
+public abstract class TestScenario
 {
-    BufferedProducerTest,
-    BurstBufferedProducerTest,
-    EventProducerTest,
-    ProcessorTest,
-    ConsumerTest
+    /// <summary>The <see cref="TestParameters"/> used to configure this test scenario.</summary>
+    internal readonly TestParameters _testParameters;
```

Comment:
If we're going to make these internal, I'd suggest making them properties.  The `_Name` pattern is generally used to imply "this is a class member used only by the class".